bugsnag-plugin-android-ndk/src/main/java/com/bugsnag/android/ndk/NativeBridge.kt

```diff
@@ -0,0 +1,164 @@
+            is AddBreadcrumb -> addBreadcrumb(msg.message, msg.type.toString(), msg.timestamp, msg.metadata)
+            NotifyHandled -> addHandledEvent()
+            NotifyUnhandled -> addUnhandledEvent()
+            PauseSession -> pausedSession()
+            is StartSession -> startedSession(msg.id, msg.startedAt, msg.handledCount, msg.unhandledCount)
+            is UpdateContext -> updateContext(msg.context ?: "")
+            is UpdateInForeground -> updateInForeground(msg.inForeground, msg.contextActivity ?: "")
+            is UpdateOrientation -> updateOrientation(msg.orientation)
+            is UpdateUserEmail -> updateUserEmail(msg.email ?: "")
+            is UpdateUserName -> updateUserName(msg.name ?: "")
+            is UpdateUserId -> updateUserId(msg.id ?: "")
+        }
+    }
+
+    private fun isInvalidMessage(msg: Any?): Boolean {
+        if (msg !is StateEvent) {
+            return true
+        }
+        if (!installed.get() && msg !is Install) {
+            warn("Received message before INSTALL: $msg")
+            return true
+        }
+
+        warn(String.format("Received NDK message %s", msg))
+        return false
+    }
+
+    private fun deliverPendingReports() {
+        lock.lock()
+        try {
+            val outDir = File(reportDirectory)
+            if (outDir.exists()) {
+                val fileList = outDir.listFiles()
+                if (fileList != null) {
+                    for (file in fileList) {
+                        deliverReportAtPath(file.absolutePath)
+                    }
+                }
+            } else {
+                warn("Report directory does not exist, cannot read pending reports")
+            }
+        } catch (ex: Exception) {
+            warn("Failed to parse/write pending reports: $ex")
+        } finally {
+            lock.unlock()
+        }
+    }
+
+    private fun handleInstallMessage(arg: Install) {
+        lock.lock()
+        try {
+            if (installed.get()) {
+                warn("Received duplicate setup message with arg: $arg")
+            }
+            val reportPath = reportDirectory + UUID.randomUUID().toString() + ".crash"
+            install(reportPath, arg.autoDetectNdkCrashes, Build.VERSION.SDK_INT, is32bit,
+                arg.appVersion ?: "", arg.buildUuid ?: "", arg.releaseStage ?: ""
+            )
+            installed.set(true)
+        } finally {
+            lock.unlock()
+        }
+    }
+
+    private fun handleAddMetadata(arg: AddMetadata) {
+        if (arg.key != null) {
+            when (val newValue = arg.value) {
+                is String -> addMetadataString(arg.section, arg.key!!, newValue)
+                is Boolean -> addMetadataBoolean(arg.section, arg.key!!, newValue)
+                is Number -> addMetadataDouble(arg.section, arg.key!!, newValue.toDouble())
+                else -> Unit
+            }
+        }
+    }
+
+    private fun warn(msg: String) {
+        Log.d("Bugsnag", msg)
```

Comment:
Shouldn't this be `Log.w`? Also, it should use the configured logger rather than Android's directly.

Reply:
It's possible to use the configured logger now that the logger PR has landed, so I'll update the changeset to use that.